augur/util_support/date_disambiguator.py

```diff
@@ -60,7 +60,32 @@ def range(self):
             resolve_uncertain_int(self.uncertain_date_components["m"], "max"),
             resolve_uncertain_int(self.uncertain_date_components["d"], "max"),
         )
-        max_date = min(max_date, datetime.date.today())
+
+        # Set default bounds.
+        lower_bound = None
+        upper_bound = datetime.date.today()
+
+        # TODO: Allow more granularity than year alone.
+        if self.min_max_year:
+            lower_bound = datetime.date(int(self.min_max_year[0]), 1, 1)
+
+            if len(self.min_max_year) == 2:
+                upper_bound = datetime.date(int(self.min_max_year[1]), 12, 31)
+
+        # Limit dates to the given bounds.
+        if lower_bound:
+            if lower_bound > max_date:
+                raise InvalidDate(self.uncertain_date, f"Not possible for date to fall within bounds '{self.min_max_year}'")
+
+            if min_date < lower_bound:
+                min_date = lower_bound
+
+        if upper_bound:
+            if upper_bound < min_date:
```

Comment:
…but here the order as-is reads well to me, for the same "picturing a number line" reason.